package/MDAnalysis/converters/accessors.py

```diff
@@ -0,0 +1,187 @@
+    property. `ag.convert_to` will return this ConverterWrapper, which can be
+    called directly with the name of the destination package as a string
+    (similarly to the old API), or through custom methods named after the
+    package (in lowercase) that are automatically constructed thanks to
+    metaclass magic.
+
+    Example
+    -------
+    The code below converts a Universe to a :class:`parmed.structure.Structure`.
+
+    .. code-block:: python
+
+        >>> import MDAnalysis as mda
+        >>> from MDAnalysis.tests.datafiles import GRO
+        >>> u = mda.Universe(GRO)
+        >>> parmed_structure = u.atoms.convert_to('PARMED')
+        >>> parmed_structure
+        <Structure 47681 atoms; 11302 residues; 0 bonds; PBC (triclinic); NOT parametrized>
+
+    You can also directly use ``u.atoms.convert_to.parmed()``
+
+    Parameters
+    ----------
+    package: str
+        The name of the package to convert to, e.g. ``"PARMED"``
+    *args:
+        Positional arguments passed to the converter
+    **kwargs:
+        Keyword arguments passed to the converter
+
+    Returns
+    -------
+    output:
+        An instance of the structure type from another package.
+
+    Raises
+    ------
+    TypeError:
+        No converter was found for the required package
+
+
+    .. versionadded:: 1.0.0
+    .. versionchanged:: 2.0.0
+        Moved the ``convert_to`` method to its own class. The old API is still
+        available and is now case-insensitive to package names, it also accepts
+        positional and keyword arguments. Each converter function can also
+        be accessed as a method with the name of the package in lowercase, i.e.
+        `convert_to.parmed()`
+    """
+    _CONVERTERS = {}
+
+    def __init__(self, ag):
+        """
+        Parameters
+        ----------
+        ag : AtomGroup
+            The AtomGroup to convert
+        """
+        self._ag = ag
+        for lib, converter_cls in _CONVERTERS.items():
+            method_name = lib.lower()
+            # makes sure we always use the same instance of the converter
+            # no matter which atomgroup instance called it
+            try:
+                converter = self._CONVERTERS[method_name]
+            except KeyError:
+                converter = converter_cls().convert
+                # store in class attribute
+                self._CONVERTERS[method_name] = converter
+            # create partial function that passes ag to the converter
+            convert = partial(converter, self._ag)
+            # copy docstring and metadata to the partial function
+            # note: it won't work with help()
+            update_wrapper(convert, converter)
+            setattr(self, method_name, convert)
+
+    def __call__(self, package, *args, **kwargs):
+        try:
+            convert = getattr(self, package.lower())
+        except AttributeError:
+            raise TypeError(f"No {package!r} converter found. Available: "
```

Comment:
This should be easy enough to test, can you add in a quick test for this @cbouy ?